### PR TITLE
[EV-4429] Update dest_domains desc, and felix DestDomainsByClient.

### PR DESCRIPTION
**Pull request by dimitri-nicolo**
Add new felix configuration variable `flowLogsDestDomainsByClient` and update the `dest_domains` description.

Product Version(s):

Issue:

[EV-4235]

Link to docs preview:

SME review:
- [x] An SME has approved this change. 

DOCS review:
- [x] A member of the docs team has approved this change.

Additional information:

Merge checklist:

- [x] Deploy preview inspected wherever changes were made 
- [x] Build completed successfully
- [x] Test have passed 

[EV-4235]: https://tigera.atlassian.net/browse/EV-4235?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: calico-enterprise/visibility/elastic/flow/datatypes.mdx [hunk 1]
```diff
@@ -28,7 +28,7 @@ The following table details the key/value pairs in the JSON blob, including thei
...
-| `dest_domains`                    | array of keywords | Top level domains associated with the destination IP. Only valid for source reported flows to destinations external to the cluster. Please note that `dest_domains: ["A"]` does not necessarily mean that this flow describes a connection to domain name `A`. It just means that the destination IP is known to be associated with domain name `A` - but it might also be associated with other domain names as well (that Calico has not yet been able to capture). |
+| `dest_domains`                    | array of keywords | This information pertains to top-level domains linked to the destination IP. It specifically applies to flows reported from the source to destinations outside the cluster. When flowLogsDestDomainsByClient is enabled, the destination domain IPs have been queried by the source IP reported in the flow log. Otherwise, when flowLogsDestDomainsByClient is not enabled, dest_domains: ["A"] doesn't guarantee that the flow corresponds to a connection with domain name A. In either case, it's possible that this destination IP is also linked to other domain names not yet captured by Calico.|
```
Comment (bmckercher123) [inline calico-enterprise/visibility/elastic/flow/datatypes.mdx:31]:
```suggestion
| `dest_domains`                    | array of keywords | Information on top-level domains linked to the destination IP. Applies specifically to flows reported from the source to destinations outside the cluster. If flowLogsDestDomainsByClient is enabled, the destination domain IPs are queried by the source IP reported in the flow log. If flowLogsDestDomainsByClient is not enabled, dest_domains: ["A"] doesn't guarantee that the flow corresponds to a connection with domain name A. Note that the destination IP may also be linked to other domain names that are not yet captured by Calico (regardless if the parameter is enabled).|
```

Comment (bmckercher123) [inline calico-enterprise/visibility/elastic/flow/datatypes.mdx]: Wondering if we are missing practical guidance. Something like this?

Information on the top-level domains linked to the destination IP. Applies to flows reported from the source to destinations outside the cluster. To find all the destination domain names for use in a DNS policy, enable `dest_domains`; this allows you to see only the domain names that the deployment connected to (not all domain names that got translated to the same IP address). So if this parameter is not enabled, dest_domains: ["A"] doesn't guarantee that the flow corresponds to a connection with domain name A. Regardless if the parameter is enabled, note that the destination IP may also linked to other domain names not yet captured by Calico.

**Review by bmckercher123 — APPROVED** [review]
LGTM. Suggestions if makes sense.

File: calico-enterprise/visibility/elastic/flow/datatypes.mdx (hunk 1 above)
Comment (dimitri-nicolo) [reply calico-enterprise/visibility/elastic/flow/datatypes.mdx]: `dest_domains` does not need to be enabled, it's just another field that the flow logs report. We want to call out that it's the `flowLogsDestDomainsByClient` that defines a different interpretation in the behaviour.

How about:
```
Find all the destination domain names for use in a DNS policy by examining `dest_domains`. The field displays information on the top-level domains linked to the destination IP. Applies to flows reported from the source to destinations outside the cluster. If `flowLogsDestDomainsByClient` is disabled, having `dest_domains`: ["A"] doesn't guarantee that the flow corresponds to a connection with domain name A. The destination IP may also linked to other domain names not yet captured by Calico.
```

OR 

```
Identify destination domain names for DNS policy considerations by examining the `dest_domains` field. This field provides information about the top-level domains associated with the destination IP. Applies to flows reported from the source to destinations external to the cluster. When `flowLogsDestDomainsByClient` is set to false, having `dest_domains`: ["A"] doesn't guarantee that the flow corresponds to a connection with the domain name "A". The destination IP might also be linked to other domain names not yet captured by Calico.
```

Comment (bmckercher123) [reply calico-enterprise/visibility/elastic/flow/datatypes.mdx]: Yes, that's is good. Perfect.

**Pull request merged**